ocsp.go

```diff
@@ -63,25 +73,41 @@ type ocspStatus struct {
 }
 
 const (
-	ocspSuccess               ocspStatusCode = 0
-	ocspNoServer              ocspStatusCode = -1
-	ocspFailedParseOCSPHost   ocspStatusCode = -2
-	ocspFailedComposeRequest  ocspStatusCode = -3
-	ocspFailedDecomposeRequst ocspStatusCode = -4
-	ocspFailedEncodeCertID    ocspStatusCode = -5
-	ocspFailedSubmit          ocspStatusCode = -6
-	ocspFailedResponse        ocspStatusCode = -7
-	ocspFailedExtractResponse ocspStatusCode = -8
-	ocspFailedParseResponse   ocspStatusCode = -9
-	ocspInvalidValidity       ocspStatusCode = -10
-	ocspRevokedOrUnknown      ocspStatusCode = -11
-	ocspMissedCache           ocspStatusCode = -12
-	ocspCacheExpired          ocspStatusCode = -13
-	ocspFailedDecodeResponse  ocspStatusCode = -14
+	ocspSuccess                ocspStatusCode = 0
+	ocspRevoked                ocspStatusCode = -1
+	ocspUnknown                ocspStatusCode = -2
+	ocspNoServer               ocspStatusCode = -3
+	ocspFailedParseOCSPHost    ocspStatusCode = -4
+	ocspFailedComposeRequest   ocspStatusCode = -5
+	ocspFailedDecomposeRequest ocspStatusCode = -6
+	ocspFailedSubmit           ocspStatusCode = -7
+	ocspFailedResponse         ocspStatusCode = -8
+	ocspFailedExtractResponse  ocspStatusCode = -9
+	ocspFailedParseResponse    ocspStatusCode = -10
+	ocspInvalidValidity        ocspStatusCode = -11
+	ocspMissedCache            ocspStatusCode = -12
+	ocspCacheExpired           ocspStatusCode = -13
+	ocspFailedDecodeResponse   ocspStatusCode = -14
 )
 
+// copied from crypto/sfocsp.go
+type certID struct {
+	HashAlgorithm pkix.AlgorithmIdentifier
+	NameHash      []byte
+	IssuerKeyHash []byte
+	SerialNumber  *big.Int
+}
+
+// cache key
+type certIDKey struct {
+	HashAlgorithm crypto.Hash
+	NameHash      string
+	IssuerKeyHash string
+	SerialNumber  string
+}
```

Comment:
Changed the cache key from a base64 encoded certID to this so that each element of key can be compared.